public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,337 @@
+    }
+  );
+
+  // Establish savedObjectLoader for all operations on vis_augment saved objects
+  const savedObjectLoader: SavedObjectLoader = getSavedFeatureAnywhereLoader();
+
+  useEffect(() => {
+    if (
+      errorGettingDetectors &&
+      !errorGettingDetectors.includes(SINGLE_DETECTOR_NOT_FOUND_MSG)
+    ) {
+      console.error(errorGettingDetectors);
+      core.notifications.toasts.addDanger(
+        typeof errorGettingDetectors === 'string' &&
+          errorGettingDetectors.includes(NO_PERMISSIONS_KEY_WORD)
+          ? prettifyErrorMessage(errorGettingDetectors)
+          : 'Unable to get all detectors'
+      );
+      setIsLoadingFinalDetectors(false);
+    }
+  }, [errorGettingDetectors]);
+
+  // Update modal state if user decides to close modal
+  useEffect(() => {
+    if (confirmModalState.isRequestingToClose) {
+      if (isLoading) {
+        setConfirmModalState({
+          ...confirmModalState,
+          isListLoading: true,
+        });
+      } else {
+        setConfirmModalState({
+          ...confirmModalState,
+          isOpen: false,
+          isListLoading: false,
+          isRequestingToClose: false,
+        });
+      }
+    }
+  }, [confirmModalState.isRequestingToClose, isLoading]);
+
+  useEffect(() => {
+    getDetectors();
+  }, []);
+
+  // Handles all changes in the assoicated detectors such as unlinking or new detectors associated
+  useEffect(() => {
+    // Gets all augmented saved objects
+    savedObjectLoader
+      .findAll()
+      .then((resp: any) => {
+        if (resp != undefined) {
+          const savedAugmentObjectsArr: ISavedAugmentVis[] = get(
+            resp,
+            'hits',
+            []
+          );
+          const curSelectedDetectors = getAssociatedDetectors(
+            Object.values(allDetectors),
+            savedAugmentObjectsArr
+          );
+          setSelectedDetectors(curSelectedDetectors);
+          setIsLoadingFinalDetectors(false);
+        }
+      })
+      .catch((error) => {
+        core.notifications.toasts.addDanger(
+          prettifyErrorMessage(`Unable to fetch associated detectors: ${error}`)
+        );
+      });
+  }, [allDetectors]);
+
+  // cross checks all the detectors that exist with all the savedAugment Objects to only display ones
+  // that are associated to the current visualization
+  const getAssociatedDetectors = (
+    detectors: DetectorListItem[],
+    savedAugmentObjects: ISavedAugmentVis[]
+  ) => {
+    // Filter all savedAugmentObjects that aren't linked to the specific visualization
+    const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+      savedAugmentObjects.filter(
+        (savedObj) => get(savedObj, 'visId', '') === embeddable.vis.id
+      );
+
+    // Map all detector IDs for all the found augmented vis objects
+    const savedAugmentDetectorsSet = new Set(
+      savedAugmentForThisVisualization.map((savedObject) =>
+        get(savedObject, 'pluginResourceId', '')
```

Comment:
`ISavedAugmentVis` interface has changed - see [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4006/files#diff-f3a7084285f53c21fad7d46ef7e11a91926e5568f3ac0c9ef9df96f015f99f2a).

Basically this will mean to change above to:
```
get(savedObject, 'pluginResource.id', '')
```